client/lecturer/app/lecturer.route.js

```diff
@@ -34,8 +34,13 @@
                 controllerAs: 'vm',
                 loginRequired: true
             })
+            .when('/lectures/:lectureId', {
+                templateUrl: 'app/lecture/lecture.html',
+                controller: 'LectureCtrl',
+                controllerAs: 'vm'
+            })
```

Comment:
You should remove the other '/lectures' route and corresponding folder. Other than that it looks good. Have not managed to test seeing comments though.